front-end/src/components/navigation/NavigationPanel.js

```diff
@@ -1,29 +1,40 @@
 import React, { Component } from 'react';
 import { NavLink } from 'react-router-dom';
 import { connect } from 'react-redux';
-import NavigationIcon from './navigation-icon/NavigationIcon';
 import styles from './navigation.module.scss';
 
-
 class NavigationPanel extends Component {
-    
+    clicked = (event) => {
+        for (let element of event.target.parentNode.children) {
+            if (element === event.target) {
+                element.className = 'active';
+            } else {
+                element.className = 'inactive';
```

Comment:
usually you just need to delete active class from other elements, using separate inactive class seems a bit weird. 
Also NavLink has a built in functionality to assign active class for active element, I'd assume it wasn't working before since you nested icon div inside a, but with a pure <a> element it should work properly by itself, no?

Reply:
Also if we do swiping between tabs in the future, there won't a click event. The basic logic in there is to compare link path to current url instead of activating by click. Ideally :)

Reply:
I put `inactive` class for a fine control over the inactive classes, since we are checking it manually now. So, in the future, if there are any cases of styling the `inactive` classes, then it will be there. 

Also, you are correct, NavLink should put an `active` class by itself. It was working even with the Nested icon component, however, it stopped working when I checked it last time. It is a known issue with NavLink that has been reported by a lot of users. I even checked by reverting `react-router-dom` to an older version, but it still doesn't work. Given the time constraint, I came up with this rather than debugging the issue with the `react-router` library.

Reply:
but elements are inactive if they don't have active class, so that's nothing but a default styling...